README.md

```diff
@@ -1512,6 +1512,53 @@ would enable the use of a `@fold` on the `adjacent_animal` vertex field of `Foo`
 }
 ```
 
+## Schema Generation
+To generate a GraphQL schema, we first map a database schema into an intermediate representation:
+the `SchemaGraph`.
+
+#### SchemaGraph
+* `SchemaGraph`
+    * Keeps track of a schema's transitive inheritance structure.
+    * Keeps track of a schema's indexes. Indexes are not used in the GraphQL schema generation and 
+      have a provisional API.
+    * Represents a database's schema through `SchemaElement` objects.
+        * `SchemaElement`
+            * Can be one of three kinds:
+                * `VertexType`, which represent schema vertices.
+                * `EdgeType`, which represent relations between vertices.
+                * `NonGraphElement`, which represents a schema element with usually no connections
+                   to other classes. A `NonGraphElement` may only have connections if it's abstract 
+                   and all its non-abstract subclasses are `VertexTypes`. 
+            * Have the following attributes:
+                * `name`, indicating the name of the class.
+                * `abstract`, indicating whether the class is abstract or not.
+                * `properties`, which are represented using GraphQLTypes.
+                * `in_connections`/`out_connections`, which represent the set of possible adjacent 
+                  `SchemaElements`.
+                    * For an `EdgeType`, its connections represent the set of allowed `VertexTypes` 
+                      and `NonGraphElements` at each end of the edge.
+                    * For `VertexType` and `NonGraphElement` objects, their connections represent
+                      the set of edges connected to them.
+            * `EdgeType` objects also have:
+                * `base_in_connection`/`base_out_connection`, which are the classes allowed at each 
+                  edge end that are the superclass of all other classes allowed at the edge end.
+
+#### Mapping an OrientDB schema to the SchemaGraph. 
+* We map classes that inherit from the base OrientDB vertex and edge classes to `VertexTypes` and 
+  `EdgeTypes` respectively. 
+* We map the remaining set of classes to `NonGraphElements`.
+
+#### Mapping the SchemaGraph to the GraphQLSchema
+* We map non-abstract `VertexTypes` to `GraphQLObjects`.
+* We map abstract `VertexType` to `GraphQLInterfaces`. If the class is a non-abstract `VertexType` 
+  we also create a `GraphQLUnion` that encompasses it and all its subclasses.
+* We represent a non-abstract `EdgeType` through `GraphQLLists` at each end type. For instance, 
+  suppose we have an `EdgeType` named Eats between Animal and Food `VertexTypes`. The 
+  corresponding Animal and Food `GraphQLObjects` will then have `out_Eats: [Food]` and 
+  `in_Eats: [Animal]` fields.
+* We map abstract `NonGraphElements` as GraphQLInterfaces iff all their non-abstract 
```

Comment:
`iff` is math jargon, try to avoid it since it limits the audience. Also, what happens if they don't fulfill this condition?